r/tools/nixlibs.R

```diff
@@ -828,10 +832,10 @@ quietly <- !env_is("ARROW_R_DEV", "true")
 
 not_cran <- env_is("NOT_CRAN", "true")
 
-if (is_release) {
+if (is_release & !test_mode) {
   VERSION <- VERSION[1, 1:3]
   arrow_repo <- paste0(getOption("arrow.repo", sprintf("https://apache.jfrog.io/artifactory/arrow/r/%s", VERSION)), "/libarrow/")
-} else {
+} else if(!test_mode) {
```

Comment:
We could also wrap this whole block in `if (!test_mode) { ... }` since we don't want to run the actual detection yet, just check the functions. Unless I'm missing something and we actually do want this whole main chunk to run?

Reply:
And just to make sure I understand: you're suggesting leaving this as is rather than making the whole block skip while being tested.

We could also suppress the message more deeply (or even in the test script) but IMHO both of those are harder to figure out what's up / reason about than stopping this one bit from running during testing and producing distracting output. But happy to change it to elsewhere if you have strong feelings about it

Reply:
This bit here is about suppressing this part, which I see in that output:

```
 *** No nightly binaries were found for version 8.0.0.9000: falling back to libarrow build from source
```

Reply:
I also just used the existing pattern so :shrug: I think it should be fine to fix it like this as further up is a block that sets the version specifically for testing and iirc we don't use the arrow repo option in any of the tests (other wise we would also see fails in ci).